src/tensorlake/function_executor/proto/function_executor.proto

```diff
@@ -161,6 +173,128 @@ message InfoResponse {
+    optional uint32 index = 3;
+}
+
+// Used for uploading a serialized object from initiating side to the accepting side.
+message UploadSerializedObjectRequest {
+    oneof message {
+        SerializedObjectManifest manifest = 1;
+        SerializedObjectChunk chunk = 2;
+    }
+}
+
+// The accepting side sends a response per request.
+message UploadSerializedObjectResponse {
+    optional Status status = 1;
+    optional SerializedObjectID id = 2;
+}
+
+// Joins the streams returned by run_task_allocations_session RPC to a session with a given ID.
+// A new session is created if a session with this ID is not known to the Function Executor.
+// Otherwise, the session streams join the existing session. Each session can only have a single
+// pair of streams joined to it.
+message OpenSessionRequest {
+    // The ID of the session to associate the session streams with.
+    optional string session_id = 1;
+}
+
+// Closes the session associated with the session streams and release all session resources.
+// Example resources: function inputs, function outputs, etc.
+message CloseSessionRequest {}
+
+message TaskAllocationInput {
+    optional string graph_invocation_id = 1;
+    optional string task_id = 2;
+    optional string allocation_id = 3;
+    optional SerializedObjectID function_input = 4;
+    optional SerializedObjectID function_init_value = 5;
+}
+
+message RunTaskAllocationsRequest {
+    // The list of task allocations to run.
+    // Currently, only one task allocation is allowed until batching gets implemented.
+    repeated TaskAllocationInput allocations = 1;
+}
+
+// A session message sent by the client to the Function Executor in run task allocations session.
+message RunTaskAllocationsSessionClientMessage {
+    oneof message {
+        OpenSessionRequest open_session_request = 1;
+        UploadSerializedObjectRequest upload_serialized_object_request = 2;
+        UploadSerializedObjectResponse upload_serialized_object_response = 3;
+        RunTaskAllocationsRequest run_task_allocations_request = 4;
+        SetInvocationStateResponse set_invocation_state_response = 5;
+        GetInvocationStateResponse get_invocation_state_response = 6;
+        CloseSessionRequest close_session_request = 7;
+    }
+}
+
+message OpenSessionResponse {
+    optional Status status = 1;
+    // True if a new session was created.
+    optional bool is_new = 2;
+}
+
+message CloseSessionResponse {
+    optional Status status = 1;
+}
+
+message TaskAllocationOutput {
+    optional string graph_invocation_id = 1;
+    optional string task_id = 2;
+    optional string allocation_id = 3;
+    optional TaskOutcomeCode outcome_code = 4;
+    optional TaskFailureReason failure_reason = 5;
+    repeated SerializedObjectID function_outputs = 6;
+    // User payload for invocation error if task failed with invocation error.
+    optional SerializedObjectID invocation_error_output = 7;
+    optional SerializedObjectID stdout = 8;
```

Comment:
This means we don't know the ID of the output / error streams until the allocation completes.

Reply:
When doing live log streaming it'd be initiated by a different message that wraps a serialized object manifest. The wrapper message will clarify which obj id is stdout and which is stderr, then the chunks for them will follow.
I think this even would be fully backward compatible.